src/engine/controls/cuecontrol.cpp

```diff
@@ -2168,11 +2169,44 @@ void CueControl::updateIndicators() {
             m_pCueIndicator->setBlinkValue(ControlIndicator::ON);
         }
     }
+
+    // Update hotcue indicators: activate the indicator if we are at the cue pos
+    // or if we passed it since the last update, else deactivate.
+    const auto prevPos = m_prevPosition.getValue();
+    const auto currPos = frameInfo().currentPosition;
+    VERIFY_OR_DEBUG_ASSERT(prevPos.isValid()) {
+        m_prevPosition.setValue(currPos);
+        return;
+    }
+
+    for (const auto& pControl : std::as_const(m_hotcueControls)) {
+        const auto cuePos = pControl->getPosition();
+        if (!cuePos.isValid()) {
+            continue;
+        }
+        if ((cuePos >= prevPos && cuePos <= currPos) ||     // forward
+                (cuePos <= prevPos && cuePos >= currPos)) { // reverse
+            pControl->setIndicator(1.0);
+        } else {
+            pControl->setIndicator(0.0);
+        }
```

Comment:
In case of a saved loop and playing reverse, the duration of the loop needs to be added.

Reply:
Good catch, I did not test looping.
And admittedly that is a tricky one. There is a seek from [loop end - x] to [loop in + y] where x + y = buffer length, so we'd need to get/store loop start/end and check loop_enabled as well as whether the hotcue is inside the loop, and that for both play directions. Uff...

Reply:
Well - I've made some experience with this topic, in #11840 and #12281

Reply:
I think I managed to catch all hotcues correctly.

Manual tests passed (with the trace output and the test mapping).
I'll update the Mixxx tests accordingly.